src/hooks/useReportIDs.tsx

```diff
@@ -0,0 +1,174 @@
+            const decision = reportAction.message?.[0]?.moderationDecision?.decision;
+
+            return {
+                reportActionID,
+                actionName,
+                errors,
+                message: [
+                    {
+                        moderationDecision: {decision},
+                    },
+                ] as Message[],
+                originalMessage,
+            };
+        })) as ReportActionsSelector;
+
+const policySelector = (policy: OnyxEntry<OnyxTypes.Policy>): PolicySelector =>
+    (policy && {
+        type: policy.type,
+        name: policy.name,
+        avatar: policy.avatar,
+        employeeList: policy.employeeList,
+    }) as PolicySelector;
+
+function ReportIDsContextProvider({
+    children,
+    /**
+     * Only required to make unit tests work, since we
+     * explicitly pass the currentReportID in LHNTestUtils
+     * to SidebarLinksData, so this context doesn't have
+     * access to currentReportID in that case.
+     *
+     * This is a workaround to have currentReportID available in testing environment.
+     */
+    currentReportIDForTests,
+}: ReportIDsContextProviderProps) {
+    const [priorityMode] = useOnyx(ONYXKEYS.NVP_PRIORITY_MODE, {initialValue: CONST.PRIORITY_MODE.DEFAULT});
+    const [chatReports] = useOnyx(ONYXKEYS.COLLECTION.REPORT, {selector: chatReportSelector});
+    const [policies] = useOnyx(ONYXKEYS.COLLECTION.POLICY, {selector: policySelector});
+    const [allReportActions] = useOnyx(ONYXKEYS.COLLECTION.REPORT_ACTIONS, {selector: reportActionsSelector});
+    const [transactionViolations] = useOnyx(ONYXKEYS.COLLECTION.TRANSACTION_VIOLATIONS);
+    const [reportsDrafts] = useOnyx(ONYXKEYS.COLLECTION.REPORT_DRAFT_COMMENT);
+    const [betas] = useOnyx(ONYXKEYS.BETAS);
+
+    const {accountID} = useCurrentUserPersonalDetails();
+    const currentReportIDValue = useCurrentReportID();
+    const derivedCurrentReportID = currentReportIDForTests ?? currentReportIDValue?.currentReportID;
+    const {activeWorkspaceID} = useActiveWorkspace();
+
+    const policyMemberAccountIDs = getPolicyEmployeeListByIdWithoutCurrentUser(policies, activeWorkspaceID, accountID);
+
+    const getOrderedReportIDs = useCallback(
+        (currentReportID?: string) =>
+            SidebarUtils.getOrderedReportIDs(
+                currentReportID ?? null,
+                chatReports,
+                betas,
+                policies,
+                priorityMode,
+                allReportActions,
+                transactionViolations,
+                activeWorkspaceID,
+                policyMemberAccountIDs,
+            ),
+        // we need reports draft in deps array for reloading of list when reportsDrafts will change
+        // eslint-disable-next-line react-hooks/exhaustive-deps
+        [chatReports, betas, policies, priorityMode, allReportActions, transactionViolations, activeWorkspaceID, policyMemberAccountIDs, reportsDrafts],
```

Comment:
Coming from https://github.com/Expensify/App/issues/54094, we decide to replace the dependency `reportsDrafts` with `Object.keys(reportsDrafts).length` to re-generate the list only when the number of drafts change, not when the draft content changes.